backend/src/test/java/com/timlenny/backend/controller/StudyplanControllerTest.java

```diff
@@ -12,29 +8,19 @@
 import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
 import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
 
-import java.util.List;
-
 @SpringBootTest
 @AutoConfigureMockMvc
 class StudyplanControllerTest {
 
     @Autowired
     MockMvc mockMvc;
-    ObjectMapper mapper = new ObjectMapper();
 
-    Topic demoTopicStart = new Topic(
-            "1", "START", List.of(new Edge("3231", "", "")), new TopicPosition(200, 200), "", "", 3, true
-    );
-    Topic demoTopicJava = new Topic(
-            "2", "Java", List.of(new Edge("32343232141", "1", "2")), new TopicPosition(300, 600), "Pending", "1852", 3, true
-    );
 
     @Test
     void whenGetAllTopics_theReturnListOfAllTopics() throws Exception {
-        String jsonObj = mapper.writeValueAsString(new Topic[]{demoTopicStart, demoTopicJava});
         mockMvc.perform(MockMvcRequestBuilders.get("/api/topic"))
-                .andExpect(MockMvcResultMatchers.status().isOk())
-                .andExpect(MockMvcResultMatchers.content().json(jsonObj));
```

Comment:
Why have you removed this? Don't you want to check whether the body contents fulfills the assertion?  Maybe you want to write two separate tests for controller's success (200) for an empty repo ([]) and non-empty one ([ demoTopicStart, demoTopicJava ])?

Reply:
Thanks for the hint! I have adjusted the tests and covered the other test cases